qfieldsync/gui/cloud_transfer_dialog.py

```diff
@@ -310,6 +316,34 @@ def show_end_page(
             self.detailedLogEndPageListView.setModel(logs_model)
             self.detailedLogEndPageListView.setModelColumn(0)
 
+    def check_localized_datasets(self):
+        localized_data_paths = (
+            QgsApplication.instance().localizedDataPathRegistry().paths()
+        )
+        if localized_data_paths:
+            localized_datasets_files = self.cloud_project.get_localized_dataset_files(
+                localized_data_paths
+            )
+            if localized_datasets_files:
+                localized_datasets_project = (
+                    self.network_manager.get_localized_datasets_project(
+                        self.cloud_project.owner
+                    )
+                )
+                if localized_datasets_project:
+                    if localized_datasets_project.id != self.cloud_project.id and (
+                        localized_datasets_project.user_role == "admin"
+                        or localized_datasets_project.user_role == "manager"
+                    ):
+                        self.localized_datasets_project = localized_datasets_project
+                        self.localized_datasets_files = localized_datasets_files
+                        reply = self.network_manager.projects_cache.get_project_files(
+                            localized_datasets_project.id
+                        )
+                        reply.finished.connect(lambda: self.prepare_project_transfer())
+                        return
```

Comment:
I am wondering if it is possible to write it without having 4 levels of nested `if`s?

Reply:
One level removed.